src/indra_cogex/sources/pubmed/__init__.py

```diff
@@ -0,0 +1,92 @@
+import csv
+import json
+import logging
+import pystow
+from indra_cogex.representation import Node, Relation
+from indra_cogex.sources.processor import Processor
+
+
+logger = logging.getLogger(__name__)
+
+resources = pystow.module("indra", "cogex", "pubmed")
+
+MESH_PMID = resources.join(name="mesh_pmids.csv")
+PMID_YEAR = resources.join(name="pmid_years_07-2021.json")
+TEXT_REFS = resources.join(name="text_refs.tsv")
+
+
+class PubmedProcessor(Processor):
+    name = "pubmed"
+
+    def __init__(
+        self,
+        mesh_pmid_path=MESH_PMID,
+        pmid_year_path=PMID_YEAR,
+        text_refs_path=TEXT_REFS,
+    ):
+        self.mesh_pmid_path = mesh_pmid_path
+        self.pmid_year_path = pmid_year_path
+        self.text_refs_path = text_refs_path
+
+    def get_nodes(self):
+        # We iterate over text refs to get the nodes and
+        # then look up the year to add as a property
+        pmid_node_type = "Publication"
```

Comment:
@cthoyt this is where the more generic node label is defined, not specific to PubMed.

Reply:
oh I totally misread this. thanks for clarifying